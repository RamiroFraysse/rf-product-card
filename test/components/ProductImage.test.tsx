import React from 'react';
import rendered from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('should render the component with the custom image', () => {
    const wrapper = rendered.create(<ProductImage img="htttps://hola.jpg" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('shold render the component with the image of the product', () => {
    const wrapper = rendered.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
