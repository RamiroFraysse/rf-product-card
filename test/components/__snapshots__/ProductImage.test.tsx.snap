// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage shold render the component with the image of the product 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage should render the component with the custom image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="htttps://hola.jpg"
/>
`;
